src/github.com/matrix-org/dendrite/syncapi/types/types.go

```diff
@@ -74,6 +74,16 @@ func NewResponse(pos StreamPosition) *Response {
 	return &res
 }
 
+// IsEmpty returns true if the response is empty, i.e. used to decided whether
+// to return the response immediately to the client or to wait for more data.
+func (r *Response) IsEmpty() bool {
+	return len(r.Rooms.Join) > 0 ||
+		len(r.Rooms.Invite) > 0 ||
+		len(r.Rooms.Leave) > 0 ||
+		len(r.AccountData.Events) > 0 ||
+		len(r.Presence.Events) > 0
+}
```

Comment:
function returns `true` if there is data although `false` is expected.

Reply:
D'oh